firewood/src/merkle/proof.rs

```diff
@@ -56,6 +56,8 @@ pub enum ProofError {
     Shale(ShaleError),
     #[error("invalid root hash")]
     InvalidRootHash,
+    #[error("failed writing the node")]
+    WriteError,
```

Comment:
```suggestion
WriteError(#[from] ObjWriteError)
```
I think that right now `ObjWriteError` doesn't actually save the original error, but it should, and when it does, it would be nice to have it propagated here. This also avoids the `map_err` with a discard and lets you use ?

This also fixes the case that someone changes `write` to return something other than `ObjWriteError` and doesn't check all the callers for their handling.